src/rules/incognia-onboarding.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @title Incognia Onboarding Rule
+ * @overview Verify if the device location behavior matches the address declared during onboarding.
+ * @gallery true
+ * @category marketplace
+ *
+ */
+async function incogniaOnboardingRule(user, context, callback) {
+  const _ = require('lodash@4.17.19')
+
+  const { IncogniaAPI } = require('@incognia/api@1.0.0')
+  const { Auth0UserUpdateUtilities } = require('@auth0/rule-utilities@0.2.0')
+
+  const { INCOGNIA_CLIENT_ID, INCOGNIA_CLIENT_SECRET } = configuration;
+
+  if (!INCOGNIA_CLIENT_ID || !INCOGNIA_CLIENT_SECRET) {
+    console.log('Missing required configuration. Skipping.');
+    return callback(null, user, context);
+  }
+
+  // For this rule to be used, please set 'incognia_onboarding_rule' as 'enabled' in your Auth0
+  // application metadata. This can be done in the advanced settings of your app.
+  const incogniaOnboardingRule = _.get(context, 'clientMetadata.incognia_onboarding_rule');
+  if (!incogniaOnboardingRule || incogniaOnboardingRule !== 'enabled') {
+    console.log('Incognia onboarding rule is not enabled for this client. Skipping');
+    return callback(null, user, context);
+  }
+
+  const installationId = _.get(context, 'request.query.installation_id');
+  if (!installationId) {
+    console.log('Missing installation_id. Skipping.');
+    return callback(null, user, context);
+  }
+
+  // User home address should be set using Auth0's Signup API for example.
+  const homeAddress = _.get(user, 'user_metadata.home_address');
```

Comment:
Might be good to allow `home_address` to be configured, in case that is already stored somewhere else in the user profile.

Reply:
Sorry, I'm not sure I understand your suggestion. Could you detail it a little bit more?

Reply:
Of course! If the user's address is already stored in a different property then a customer would either need to duplicate it to this property or the couldn't use this Rule. Instead, we could have something like `configuration. INCOGNIA_HOME_ADDRESS_PROP` that takes a string in a similar format. A tenant admin could set that to, say, 'app_metadata.address' and it would work the same way. Just default to the hard-coded string you have here already. Something like:

```
const homeAddressProp = configuration. INCOGNIA_HOME_ADDRESS_PROP || 'user_metadata.home_address';
const homeAddress = _.get(user, homeAddressProp);

Reply:
Oh! Sure! Makes sense. Done!